newsfragments/959.feature.rst

```diff
@@ -0,0 +1,8 @@
+If you have a `~trio.abc.ReceiveStream` object, you can now use
+``async for data in stream: ...`` instead of calling
+`~trio.abc.ReceiveStream.receive_some` repeatedly. And the best part
+is, it automatically checks for EOF for you, so you don't have to.
```

Comment:
I might rephrase this ("And the best part is ...") as something like "Receiving EOF will cleanly terminate the `async for` loop, so you don't have to check for it explicitly."